starter_code/elevator.js

```diff
@@ -1,19 +1,68 @@
 class Elevator {
-  constructor(){
-    this.floor      = 0;
-    this.MAXFLOOR   = 10;
-    this.requests   = [];
+  constructor() {
+    this.floor = 0;
+    this.MAXFLOOR = 10;
+    this.requests = [];
+    this.waitingList = [];
+    this.passengers = [];
+    this.direction = 'up';
   }
 
-  start() { }
-  stop() { }
-  update() { }
-  _passengersEnter() { }
-  _passengersLeave() { }
-  floorUp() { }
-  floorDown() { }
-  call() { }
-  log() { }
+  start() {
+    var interval = setInterval(() => {
+      this.update();
+    }, 1000);
+  }
+  stop() {
+    clearInterval(interval);
+  }
+  update() {
+    if (this.direction === 'up')
+      this.floorUp();
+    else if (this.direction === 'down')
+      this.floorDown();
+    this.log();
+    _passengersEnter();
```

Comment:
this is failing because it should be `this._passengerEnter()` and `this._passengerLeave()`